src/core/waitForCoreSync.js

```diff
@@ -5,16 +5,26 @@ const wait = require('../util/wait');
  *
  * @typedef {waitForCoreSync}
  * @param {CoreService} coreService
+ * @param {task} task
  * @return {Promise<void>}
  */
-async function waitForCoreSync(coreService) {
+async function waitForCoreSync(coreService, task) {
```

Comment:
It’s better to pass a callback and call it with data so we can decouple the listr and this function.

Reply:
OK, I think I did it right. Testing this was strange - previously the `Sync Core with network` step would wait for core to sync, but passing the callback function caused `waitForCoreSync()` to return immediately, even though it wasn't complete yet. I had to add `await` to the function call. Why?